routers/web/repo/view.go

```diff
@@ -142,10 +142,30 @@ func renderDirectory(ctx *context.Context, treeLink string) {
 		return
 	}
 
+	renderDirectoryDefaultFile(ctx, entries, treeLink)
+}
+
+// renderDirectoryDefaultFile render the default file when render a root directory or a sub directory
+func renderDirectoryDefaultFile(ctx *context.Context, entries git.Entries, treeLink string) {
 	if ctx.Repo.TreePath != "" {
 		ctx.Data["Title"] = ctx.Tr("repo.file.title", ctx.Repo.Repository.Name+"/"+path.Base(ctx.Repo.TreePath), ctx.Repo.RefName)
 	}
 
+	// Check permission to add or upload new file.
+	if ctx.Repo.CanWrite(unit_model.TypeCode) && ctx.Repo.IsViewBranch {
+		ctx.Data["CanAddFile"] = !ctx.Repo.Repository.IsArchived
+		ctx.Data["CanUploadFile"] = setting.Repository.Upload.Enabled && !ctx.Repo.Repository.IsArchived
+	}
```

Comment:
this block seems not related to `renderDirectoryDefaultFile`, but to `renderDirectory`.